torchgeo/datasets/loveda.py

```diff
@@ -0,0 +1,303 @@
+
+        images = sorted(images)
+
+        if self.split != "test":
+            masks = [image.replace("images_png", "masks_png") for image in images]
+            files = [
+                dict(image=image, mask=mask) for image, mask, in zip(images, masks)
+            ]
+        else:
+            files = [dict(image=image) for image in images]
+
+        return files
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            tensor: the loaded image
+
+        """
+        filename = os.path.join(path)
+        with Image.open(filename) as img:
+            array = np.array(img.convert("RGB"))
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            # Convert from HxWxC to CxHxW
+            tensor = tensor.permute((2, 0, 1))
+            return tensor
+
+    def _load_target(self, path: str) -> Tensor:
+        """Load a single mask corresponding to image.
+
+        Args:
+            path: path to the mask
+        Returns:
+            tensor: the mask of the image
+
+        """
+        filename = os.path.join(path)
+        with Image.open(filename) as img:
+            array = np.array(img.convert("L"))
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            tensor = tensor.to(torch.long)  # type: ignore[attr-defined]
+            return tensor
+
+    def _check_integrity(self) -> bool:
+        """Check the integrity of the dataset structure.
+
+        Returns:
+            True if the dataset directories and split files are found, else False
+
+        """
+        for s in self.scene_paths:
+            if not os.path.exists(s):
+                return False
+
+        return True
+
+    def _download(self) -> None:
+        """Download the dataset and extract it.
+
+        Raises:
+            AssertionError: if the checksum of split.py does not match
+
+        """
+        if self._check_integrity():
+            print("Files already downloaded and verified")
+            return
+
+        download_and_extract_archive(
+            self.url,
+            self.root,
+            filename=self.filename,
+            md5=self.md5 if self.checksum else None,
+        )
+
+    def plot(self, sample: Dict[str, Tensor], suptitle: Optional[str] = None) -> Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returne by :meth:`__getitem__`
+            suptitle: optional suptitle to use for figure
+        """
+        if self.split != "test":
+            images, masks = sample["image"], sample["mask"]
+            ncols = 2
+        else:
+            images = sample["image"]
+            ncols = 1
+
+        fig, axs = plt.subplots(nrows=1, ncols=ncols, figsize=(10, ncols * 10))
+
+        if self.split != "test":
+
+            axs[0].imshow(images[0, :, :, :].permute(1, 2, 0))
```

Comment:
The plot method expects a batch dim to exist but your dataset __getitem__ doesn't return data with a batch dim.